tracks/rust/exercises/bob/mentoring.md

```diff
@@ -0,0 +1,140 @@
+### Concepts
+
+- [bool](https://doc.rust-lang.org/std/primitive.bool.html)
+- [char](https://doc.rust-lang.org/std/char/index.html)
+- [match statement](https://doc.rust-lang.org/1.2.0/book/match.html)
+- [variables](https://doc.rust-lang.org/book/ch03-01-variables-and-mutability.html)
+- [closures](https://doc.rust-lang.org/book/ch13-01-closures.html)
+- [&str](https://doc.rust-lang.org/std/primitive.str.html)
+- [Iterators](https://doc.rust-lang.org/std/iter/trait.Iterator.html)
+- [Iterator Methods](https://doc.rust-lang.org/std/iter/trait.Iterator.html#provided-methods)
+- [constant](https://doc.rust-lang.org/std/keyword.const.html)
+
+### Reasonable Solutions
+
+A reasonable solution should:
+
+- Use constants
+- Use variables
+- Use `match` statements
+- Check if the input string is empty and return the "Sure" constant
+before doing any additional processing
+- Use simple methods from the standard library
+- Not use the regex crate
+- Not use separate functions
```

Comment:
I disagree here. Modularization is somewhat style-dependent, but there's not real disadvantage to putting things into functions, generally speaking. I'd really not like to push that onto people who're new to the language and try to organize things in a way that fits them, if it does not come with a definite disadvantage. Of course, people should not make useless separations that only add lines like `fn is_empty(s: &str) { s.is_empty() }`, but that's much less general than what you've written here.

Reply:
This is a generalisation because, for the most part, when they do create separate functions, they are overcomplicating things. This guidance is so mentors can help direct them to a more concise and idiomatic way of thinking :)

Here's just one example: `fn is_question(message: &str) -> bool {
    message.ends_with("?")
}`

This can simply be `let is_question: bool = trimmed_message.ends_with("?");`

And, yes, I have seen `.is_empty()` in a separate function. Most, if not all, of the separate functions are "useless separations" :)

Reply:
> This is a generalisation because, for the most part, when they do create separate functions, they are overcomplicating things.

I suggest avoiding generalisations in mentor notes. Some mentors are going to take it at face value and force students to do this, if you phrase it this way.

> And, yes, I have seen `.is_empty()` in a separate function. Most, if not all, of the separate functions are "useless separations" :)

That's not true. A separation like my example is useless because it doesn't actually modularize something, save lines or adds clarity for code. Look at the clock exercise, where people sometimes factor out the `modulo` function, which is only one line. This would not be "useless", although it adds more lines.

Reply:
@TheDarkula My 2 cents here:

"Not use separate functions" is too strong of a wording but I understand what you're trying to accomplish. I would suggest you remove it here, but we can make subsequent edits to this note adding _examples_ why the separated out function is unnecessary. 

```suggestion
```